crates/wasi-common/yanix/src/sys/linux/dir.rs

```diff
@@ -41,6 +41,6 @@ pub(crate) unsafe fn iter_impl(dir: &Dir) -> Option<Result<EntryImpl>> {
             None
         }
     } else {
-        Some(Ok(EntryImpl(*dirent)))
+        Some(Ok(EntryImpl(unsafe { *dirent })))
```

Comment:
Should we use [std::ptr::read_unaligned](https://doc.rust-lang.org/std/ptr/fn.read_unaligned.html) here instead of simple pointer dereferencing, since we're doing a clean up of those here anyway?

Reply:
This pointer comes from `libc::readdir64`, which we can assume will give us an aligned pointer. In the code where `write_unaligned` is needed, we're writing into a `Vec<u8>` buffer which isn't guaranteed to be aligned.

Reply:
Right, got it, thanks!